v1/transform.go

```diff
@@ -0,0 +1,344 @@
+					Minimum:      r.Properties.Minimum,
+					Maximum:      r.Properties.Maximum,
+					DefaultValue: r.Properties.DefaultValue,
+					Mask:         r.Properties.Mask,
+					Shift:        r.Properties.Shift,
+					Scale:        r.Properties.Scale,
+					Offset:       r.Properties.Offset,
+					Base:         r.Properties.Base,
+					Assertion:    r.Properties.Assertion,
+					MediaType:    r.Properties.MediaType,
+				},
+				Units: v1Models.Units{
+					Type:         "String",
+					ReadWrite:    common.ReadWrite_R,
+					DefaultValue: r.Properties.Units,
+				},
+			},
+		}
+		if r.Properties.ValueType == common.ValueTypeFloat32 || r.Properties.ValueType == common.ValueTypeFloat64 ||
+			r.Properties.ValueType == common.ValueTypeFloat32Array || r.Properties.ValueType == common.ValueTypeFloat64Array {
+			resources[i].Properties.Value.FloatEncoding = "eNotation"
+		}
+	}
+
+	return resources
+}
+
+func toV1Tags(tags map[string]interface{}) map[string]string {
+	dto := make(map[string]string)
+	for k, v := range tags {
+		dto[k] = fmt.Sprintf("%v", v)
+	}
+	return dto
+}
+
+func toV1Attribute(attributes map[string]interface{}) map[string]string {
+	dto := make(map[string]string)
+	for k, v := range attributes {
+		dto[k] = fmt.Sprintf("%v", v)
+	}
+	return dto
+}
+
+const (
+	ResourceOperationGet = "get"
+	ResourceOperationSet = "set"
+)
+
+func toV1DeviceCommands(deviceCommands []models.DeviceCommand) []v1Models.ProfileResource {
+	commands := make([]v1Models.ProfileResource, len(deviceCommands))
+	for i, c := range deviceCommands {
+		commands[i] = v1Models.ProfileResource{
+			Name: c.Name,
+		}
+		if strings.Contains(c.ReadWrite, common.ReadWrite_R) {
+			commands[i].Get = toV1Operation(ResourceOperationGet, c.ResourceOperations)
+		}
+		if strings.Contains(c.ReadWrite, common.ReadWrite_W) {
+			commands[i].Set = toV1Operation(ResourceOperationSet, c.ResourceOperations)
+		}
+	}
+
+	return commands
+}
+
+func toV1Operation(op string, resourceOperations []models.ResourceOperation) []v1Models.ResourceOperation {
+	operations := make([]v1Models.ResourceOperation, len(resourceOperations))
+	for i, ro := range resourceOperations {
+		valueMappings := ro.Mappings
+		if op == ResourceOperationSet && len(ro.Mappings) != 0 {
+			valueMappings = reverseMapKeyValue(ro.Mappings)
+		}
+		operations[i] = v1Models.ResourceOperation{
+			Index:          strconv.Itoa(i),
+			Operation:      op,
+			DeviceResource: ro.DeviceResource,
+			Mappings:       valueMappings,
+		}
+	}
+	return operations
+}
+
+func reverseMapKeyValue(mappings map[string]string) map[string]string {
+	valueMappings := make(map[string]string, len(mappings))
+	for k, v := range mappings {
+		valueMappings[v] = k
+	}
+	return valueMappings
+}
+
+func toV1CoreCommand(deviceCommands []models.DeviceCommand) []v1Models.Command {
+	commands := make([]v1Models.Command, len(deviceCommands))
+	for i, c := range deviceCommands {
+		commands[i] = v1Models.Command{
```

Comment:
we should only create CoreCommand when the isHidden is false
add the same loop for DeviceResource because it can be CoreCommand if the isHidden is false

Reply:
if the DeviceCommand has the same name with DeviceResource, DeviceCommand take priority.
Avoid duplicate name in CoreCommand.